_patches/int8_linear_kernel_mps.py

```
@@ -340,6 +344,110 @@ def _try_int8_kernel_forward(self, input):
+        elif v in ("1", "on", "true"):
+            _DEQUANT_MODE = True
+        else:
+            try:
+                import psutil
+                _DEQUANT_MODE = psutil.virtual_memory().total >= 48 * (1 << 30)
+            except Exception:
+                _DEQUANT_MODE = False
+    return _DEQUANT_MODE
+
+
+def _maybe_dequant_weight(self, input):
```

Ruff (0.16.1)

[warning] Too many return statements (8 > 6)

(PLR0911)